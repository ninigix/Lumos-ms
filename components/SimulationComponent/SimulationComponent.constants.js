export const LABELS = {
  ON: "Light on",
  OFF: "Light off"
};
