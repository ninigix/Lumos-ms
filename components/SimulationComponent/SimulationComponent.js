import React from 'react';
import {StyleSheet, Text, View, TouchableOpacity} from "react-native";
import MyText from "../MyText/MyText";
import {Button} from "react-native-elements";
import {RkButton, RkChoice} from "react-native-ui-kitten";

// TODO: add icon
const SimulationComponent = ({date,  hour, roomName, isLightOn}) => {
    return (
        <View style={styles.dateComponentWrapper}>
            <View style={styles.dateComponentGrey}>
                <MyText isBold> {roomName} {isLightOn ? "Light on" : "Light off"}</MyText>
            </View>
            <View style={styles.dateComponentSideButton}>
                <MyText>{date}</MyText>
                <MyText>{hour}</MyText>
            </View>
        </View>)
};

export default SimulationComponent;

const styles = StyleSheet.create({
    dateComponentGrey: {
        flex: 3,
        justifyContent: 'center',
        alignItems: 'center',
        borderTopLeftRadius: 4,
        borderBottomLeftRadius: 4
    },
    dateComponentSideButton: {
        flex: 1,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        borderLeftWidth: 1,
        borderLeftColor: 'grey',
        minWidth: '15%',
    },
    dateComponentWrapper: {
        display: 'flex',
        height: 50,
        borderRadius: 4,
        borderWidth: 0.5,
        borderColor: '#d6d7da',
        flexDirection: 'row',
        margin: 15,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
    },
});