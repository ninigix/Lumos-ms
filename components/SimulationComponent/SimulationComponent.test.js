import renderer from 'react-test-renderer';
import React from 'react';

import SimulationComponent from './SimulationComponent';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <SimulationComponent />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
