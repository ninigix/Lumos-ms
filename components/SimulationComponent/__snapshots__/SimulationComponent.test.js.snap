// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "display": "flex",
      "flexDirection": "row",
      "height": 70,
      "margin": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomLeftRadius": 4,
        "borderTopLeftRadius": 4,
        "flex": 5,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EBEBE4",
            "borderRadius": 4,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "margin": 20,
            "width": 40,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "raleway-light",
              "fontSize": 16,
            }
          }
        >
          Light off
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "raleway-bold",
              "fontSize": 16,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "minWidth": "15%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EBEBE4",
          "borderBottomRightRadius": 4,
          "borderTopRightRadius": 4,
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;
